docs/fundamentals/apicompat/preview-apis.md

```diff
@@ -0,0 +1,108 @@
+---
+title: Preview APIs
+description: Provides an understand how APIs can be marked as preview.
+ms.date: 4/4/2024
+---
+
+# Preview APIs
+
+The .NET platform prides itself with taking compatibility very seriously. As
+such, the the library ecosystem tends to avoid making breaking changes,
+especially with respect to the API.
+
+Nonetheless, when designing APIs it's important to collect feedback from users
+and change the API based on their feedback. To avoid surprises, it's important
+that users understand which APIs are considered stable and which ones are still
+in active development and might change.
+
+There are multiple ways an API can express to be in preview form:
+
+* The entire component is considered preview:
+    - Exposed in a preview release of the .NET runtime
+    - Exposed in a prerelease NuGet package
+* An otherwise stable component specifically marks certain APIs as preview:
+    - Marked with <xref:System.Runtime.Versioning.RequiresPreviewFeaturesAttribute>
+    - Marked with <xref:System.Diagnostics.CodeAnalysis.ExperimentalAttribute>
+
+This document explains how to chose between these and how each of them work.
+
+## .NET runtime previews
+
+For the exception of RCs with a go-live license, preview versions of the .NET
+runtime and SDK aren't supported.
+
+As such, any APIs added as part of a .NET preview are considered subject to
+change, based on feedback the previews receive. In order to use a .NET runtime
+preview, you need to explicitly target a newer framework version. By doing so
+you implicitly express consent to consume APIs that might change.
+
+## Prerelease NuGet packages
+
+NuGet packages can either be stable or marked as prerelease, that is the package
+has a prerelease suffix. For example, `System.Text.Json 9.0.0-preview.2.24128.5`
+has a prerelease suffix of `preview.2.24128.5`.
+
+Package authors generally don't support prerelease packages and use as a means
+to collect feedback from early adopters.
+
+When installing packages, either via the CLI or via UI, you generally have to
+explicitly indicate whether you want to install prerelease, again implicitly
+expressing consent to consume APIs that might change.
+
+## `RequiresPreviewFeaturesAttribute`
+
+Since .NET 6, the platform includes <xref:System.Runtime.Versioning.RequiresPreviewFeaturesAttribute>.
+
+This is used for APIs that require preview behaviors across the stack, including
+the runtime, the C# compiler, and libraries. When consuming APIs marked with
+this attribute you will receive a build error unless your project sets
+`<EnablePreviewFeatures>true</EnablePreviewFeatures>`. Doing so will also set
+`<LangVersion>Preview</LangVersion>`, allowing the use of preview language
+features.
+
+We used that in .NET 6 to test out generic math, which required static interface
```

Comment:
```suggestion
The .NET team used the <xref:System.Runtime.Versioning.RequiresPreviewFeaturesAttribute> attribute in .NET 6 to test out generic math, because it required static interface
```